src/osd/OSDMap.cc

```diff
@@ -1118,8 +1122,13 @@ void OSDMap::set_max_osd(int m)
   osd_xinfo.resize(m);
   osd_addrs->client_addr.resize(m);
   osd_addrs->cluster_addr.resize(m);
-  osd_addrs->hb_back_addr.resize(m);
-  osd_addrs->hb_front_addr.resize(m);
+  if (require_osd_release >= CEPH_RELEASE_MIMIC) {
```

Comment:
The "then" branch should be swapped with the "else" branch. 
We need to populate the `hb_(back|front)_addr` vector when `require_osd_release < CEPH_RELEASE_MIMIC`

This will fix the segfault in the osdmap unit test. Maybe it will also fix the the failure in the teuthology rados suite.